docs/source/project_documentation/rest-api.md

```diff
@@ -0,0 +1,199 @@
+
+```python
+get("/files/context")
+```
+HTTPStatus: `200 OK`
+This command reads and returns a list of files in the current `LanguageContext`.
+
+```python
+get("/files/available")
+```
+HTTPStatus: `200 OK`
+This command reads and returns a list of files in the workspace available for import into the active context. The list of files returned does not include files already in the context.
+
+```python
+get("/file", file_uri: str)
+```
+HTTPStatus: `200 OK`
+This command accepts file uri as an argument and returns the requested file.
+
+```python
+post("/files/import", file_models: list[FilePathModel])
+```
+HTTPStatus: `NO_CONTENT`
+This command imports a list of files into the `LanguageContext`.
+
+```python
+put("/file", rename_request: FilePathRenameModel)
+```
+HTTPStatus: `NO_CONTENT`
+This command renames a file, or updates its uri.
+
+```python
+delete("/file", file_uri: str)
+```
+HTTPStatus: `NO_CONTENT`
+This command removes the file from the active `LanguageContext`.
+
+```python
+get("/definitions")
+```
+HTTPStatus: `200 OK`
+This command returns a list of definitions in the active `LanguageContext`.
+
+```python
+get("/definition", name: str)
+```
+HTTPStatus: `200 OK`
+This command accepts a definition name as a string, and returns a list of definitions with that name.
+
+```python
+post("/definition", definition_model: DefinitionModel)
+```
+HTTPStatus: `NO_CONTENT`
+This command Adds a definition to the active `LanguageContext`.
+
+```python
+post("/definitions", definition_models: list[DefinitionModels])
+```
+HTTPStatus: `NO_CONTENT`
+This command Adds a list of definitions the the active `LanguageContext`.
+
+```python
+put("/definition", definition_model: DefinitionModel)
+```
+HTTPStatus: `NO_CONTENT`
+This command updates a definition in the active `LanguageContext`.
+
+```python
+delete("/definition", name: str)
+```
+HTTPStatus: `NO_CONTENT`
+This command removes a definition from the active `LanguageContext`.
+
+```python
+get("/context/schema", key: str)
+```
+HTTPStatus: `OK`
+This command takes in a root key and returns the corresponding root YAML schema.
+
+```python
+get("/context/root_keys")
+```
+HTTPStatus: `OK`
+This command returns a list of root keys from the active `LanguageContext`.
+
+```python
+get("/command")
+```
+HTTPStatus: `OK`
+Returns a list of all available commands.
+
+```python
+post("/command", command_request: CommandRequestModel)
+```
+HTTPStatus: `OK`
+This command executes the given AaC command.
+
+## Gen-openapi-spec Command
+
+The `gen-openapi-spec` command returns a JSON file containing the open API schema. Below is an example of a successful run of the command.
+
+```bash
+$ aac gen-openapi-spec ./output_directory
+Successfully wrote the OpenAPI spec to ./output_directory/AaC_OpenAPI_Schema.json.
+```
```

Comment:
Should include a screen capture of showing the file now existing at the provided output directory location.

Reply:
Also, on my end at least, looks like it is not creating the output directory as expected for when you pass it in. It only wants to work with an existing directory it seems.